### PR TITLE
XW-2448 | Portable Infoboxes: fixing bugs with <gallery> tag used as image

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/XW-2448
https://wikia-inc.atlassian.net/browse/XW-2491
https://wikia-inc.atlassian.net/browse/XW-2211

Using gallery data from parser instead of parsing <gallery> tag by regexps.

//cc @Wikia/x-wing  @macbre

### CODE REVIEW

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php [hunk 1]
```diff
@@ -142,14 +142,16 @@ protected function renderGroup( $groupData ) {
...
 	protected function renderImage( $data ) {
 		$helper = $this->getImageHelper();
+
+		$data = $this->filterImageData( $data );
 		$images = [ ];
 
-		for ( $i = 0; $i < count( $data ); $i++ ) {
-			$data[$i]['context'] = null;
-			$data[$i] = $helper->extendImageData( $data[$i], $this->imagesWidth );
+		foreach ( $data as &$dataItem ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: Changes on currently iterated array might have unpredicted consequences. Can you do this differently?

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php [hunk 2]
```diff
@@ -169,6 +171,20 @@ protected function renderImage( $data ) {
 		return $this->render( $templateName, $data );
 	}
 
+	protected function filterImageData( $data ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: I guess this should be a `private` method. Is there a reason for it to be protected? Could you move it below `render` functions as well?

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php [hunk 3]
```diff
@@ -169,6 +171,20 @@ protected function renderImage( $data ) {
...
+			return !empty( $item['caption'] );
+		});
+
+		if ( !empty( $dataWithCaption ) ) {
+			return $dataWithCaption;
+		} elseif ( !empty( $data ) ) {
+			return [ $data[0] ];
+		}
+
+		return [];
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: Please limit number of return statements it this function.

Comment (latata) [reply extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: +1

**Review by idradm — APPROVED** [review]

**Review by latata — APPROVED** [review]

**Review by latata — APPROVED** [review]

File: extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php [hunk 4]
```diff
@@ -32,22 +32,16 @@ public static function onImageServingCollectImages( &$imageNamesArray, $articleT
...
-	 * @param $parser
-	 * @param $frame
+	 * Store data of all galleries in article to handle images in infoboxes
 	 *
+	 * @param $marker
+	 * @param WikiaPhotoGallery $gallery
 	 * @return bool
 	 */
-	public static function onParserTagHooksBeforeInvoke( $name, $marker, $content, $attributes, $parser, $frame ) {
-		if ( $name === self::PARSER_TAG_GALLERY ) {
```
Comment (kvas-damian) [inline extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php:47]: `PARSER_TAG_GALLERY` is not used now. Can we remove it?

Comment (rybmat) [reply extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php:47]: done

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**